numpy/random/_common.pyx

```diff
@@ -218,6 +218,21 @@ cdef np.ndarray int_to_array(object value, object name, object bits, object uint
     return out
 
 
+cdef int validate_output_shape(iter_shape, np.ndarray output) except -1:
+    cdef np.npy_intp *shape, ndim, i
+    cdef bint error
+    dims = np.PyArray_DIMS(output)
+    ndim = np.PyArray_NDIM(output)
+    output_shape = tuple((dims[i] for i in range(ndim)))
+    if iter_shape != output_shape:
+        raise ValueError(
+            f"Output size {output_shape} is not compatible with broadcast "
+            f"dimensions of inputs {iter_shape}."
+        )
+        return -1
```

Comment:
Are both these lines executed? This page suggests that the exception is ignored:

https://notes-on-cython.readthedocs.io/en/latest/misc.html#cdef-exceptions-and-the-return-type

Reply:
This suggests that the `return -1` is not necessary:

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#error-return-values

Reply:
I didn't have it in an earlier draft and I was getting reports of errors bein raised but not terminating the execution. 

I have line coverage on randomgen and it looks like it isn't necessary (or run):

https://coveralls.io/builds/31259701/source?filename=randomgen%2Fcommon.pyx#L515

Reply:
Fixed this.